py_zipkin/zipkin.py

```diff
@@ -169,7 +170,17 @@ def __init__(
     def __call__(self, f):
         @functools.wraps(f)
         def decorated(*args, **kwargs):
-            with self:
+            with zipkin_span(
+                service_name=self.service_name,
+                span_name=self.span_name,
+                zipkin_attrs=self.zipkin_attrs,
```

Comment:
I do think that the decorator accepts more arguments than it should, but that'll be a backwards incompatible change. We can open up another issue and address it there